shared/browser-app.cpp

```diff
@@ -105,11 +108,45 @@ bool BrowserApp::OnProcessMessageReceived(CefRefPtr<CefBrowser> browser,
 		ExecuteJSFunction(browser, "onVisibilityChange", arguments);
 		return true;
 	}
-	else if (message->GetName() == "SceneChange") {
+	else if (message->GetName() == "DispatchJSEvent") {       
+		CefRefPtr<CefV8Context> context = browser->GetMainFrame()->GetV8Context();
+
+		context->Enter();
+
+		CefRefPtr<CefV8Value> globalObj = context->GetGlobal();
+
+		json_t *json = json_object();
+        
+		if (args->GetSize() > 1) {
+			json_error_t error;
+
+			json_object_set_new(json, "detail", json_loads(args->GetString(1).ToString().c_str(), 0, &error));
+		}
+
+		char *jsonString = json_dumps(json, 0);
+
+		std::string script = fmt::format(
+			"new CustomEvent('{}', {});", 
+			args->GetString(0).ToString(),
+			jsonString);
+
+		free(jsonString);
+
+		CefRefPtr<CefV8Value> returnValue;
+		CefRefPtr<CefV8Exception> exception;
+
+		// Create the CustomEvent object
+		// We have to use eval to invoke the new operator
+		context->Eval(script, returnValue, exception);
```

Comment:
This method's arguments have been changed in CEF branch 2840, should now be:

`context->Eval(script, browser->GetMainFrame()->GetURL() ,0, returnValue, exception);`